tests/unit/element-transformer/ElementTransformer.ts

```diff
@@ -0,0 +1,292 @@
+import elementTransformer from '../../../src/element-transformer/index';
+import * as ts from 'typescript';
+
+const { describe, it } = intern.getInterface('bdd');
+const { assert } = intern.getPlugin('chai');
+
+describe('element-transformer', () => {
+	it('normalizes string arg to object', () => {
+		const source = `
+		class WidgetBase<T> {}
+		interface DojoInputProperties {}
+		@customElement("dojo-input")
+		export class DojoInput extends WidgetBase<DojoInputProperties> {
+		}`;
+		const expected = `
+		class WidgetBase<T> {}
+		interface DojoInputProperties {}
+		@customElement({ tag: "dojo-input" })
+		export class DojoInput extends WidgetBase<DojoInputProperties> {
+		}`;
+		assertCompile(source, expected, (program) => ({
+			before: [elementTransformer(program)]
+		}));
+	});
+
+	it('extracts attributes', () => {
+		const source = `
+		class WidgetBase<T> {}
+		interface DojoInputProperties {
+			a: string;
+			b: string;
+		}
+		@customElement({ tag: "dojo-input" })
+		export class DojoInput extends WidgetBase<DojoInputProperties> {
+			public a = "1";
```

Comment:
adding these public class properties to these tests are confusing me, what relevance do they have?

Reply:
None at the moment, I had misread the requirements and was pulling the public properties off the class instead of the interface.